From linux-kernel-owner+greg=40kroah.com-S932139AbWIXW45@vger.kernel.org Sun Sep 24 15:58:24 2006
From: Eric Sesterhenn <snakebyte@gmx.de>
To: linux-kernel@vger.kernel.org
Date: Mon, 25 Sep 2006 00:56:53 +0200
Subject: pciehp: Remove unnecessary check in pciehp_ctrl.c
Message-Id: <1159138613.9062.17.camel@alice>

this was spotted by coverity (cid #819). We dereference p_slot
earlier in the function, and i found no way it could become NULL
anywhere.

Signed-off-by: Eric Sesterhenn <snakebyte@gmx.de>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/pci/hotplug/pciehp_ctrl.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/pci/hotplug/pciehp_ctrl.c
+++ gregkh-2.6/drivers/pci/hotplug/pciehp_ctrl.c
@@ -797,8 +797,7 @@ int pciehp_enable_slot(struct slot *p_sl
 		p_slot->hpc_ops->get_latch_status(p_slot, &getstatus);
 	}
 
-	if (p_slot)
-		update_slot_info(p_slot);
+	update_slot_info(p_slot);
 
 	mutex_unlock(&p_slot->ctrl->crit_sect);
 	return rc;
